### PR TITLE
Add default webhook shift message before install

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
This is related to https://github.com/istio/istio/issues/48643.

There's an inconsistent behavior between:
```
default installation -> then do a revisioned installation # this will use default's default webhook

revisioned installation -> default installation # this will change the default webhook from revisioned to default
```

Instead of changing the default behaviors described in https://istio.io/latest/docs/setup/upgrade/canary/#default-tag, which may affect existing users, I think adding a warning about the default webhook shifting would be good.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by hanxiaop** [comment]
Sample out about a revisioned installation following a default installation:
```
➜ istioctl install -f out/yamls/overlay.yaml --revision=test-rev
This will install the Istio 1.21.0 "ambient" profile (with components: Istio core, Istiod, CNI, and Ztunnel) into the cluster. Proceed? (y/N) y
✔ Istio core installed                                                                                                                                                                
✔ Istiod installed                                                                                                                                                                    
✔ CNI installed                                                                                                                                                                       
✔ Ztunnel installed                                                                                                                                                                   
✔ Installation complete                                                                                                                                                               Made this installation the default for injection and validation.
➜ istioctl install -f out/yamls/overlay.yaml                    
This installation will make default injection and validation pointing to the default revision, and originally it was pointing to the revisioned one.
This will install the Istio 1.21.0 "ambient" profile (with components: Istio core, Istiod, CNI, and Ztunnel) into the cluster. Proceed? (y/N) 
```

**Comment by hanxiaop** [comment]
Sample out about a default installation following a revisioned installation (should have no message since the default installation doesn't change):
```
➜  istioctl install -f out/yamls/overlay.yaml
This will install the Istio 1.21.0 "ambient" profile (with components: Istio core, Istiod, CNI, and Ztunnel) into the cluster. Proceed? (y/N) y
✔ Istio core installed                                                                                                                                                                
✔ Istiod installed                                                                                                                                                                    
✔ CNI installed                                                                                                                                                                       
✔ Ztunnel installed                                                                                                                                                                   
✔ Installation complete                                                                                                                                                               Made this installation the default for injection and validation.
➜  istioctl install -f out/yamls/overlay.yaml --revision=test-rev
This will install the Istio 1.21.0 "ambient" profile (with components: Istio core, Istiod, CNI, and Ztunnel) into the cluster. Proceed? (y/N)
```

File: operator/cmd/mesh/install.go [hunk 1]
```diff
@@ -179,6 +180,11 @@ func Install(kubeClient kube.CLIClient, rootArgs *RootArgs, iArgs *InstallArgs,
 	// Ignore the err because we don't want to show
 	// "no running Istio pods in istio-system" for the first time
 	_ = detectIstioVersionDiff(p, tag, ns, kubeClient, iop)
+	exists := revtag.PreviousInstallExists(context.Background(), kubeClient.Kube())
+	err = detectDefaultWebhookChange(p, kubeClient, iop, exists)
+	if err != nil {
+		return err
```
Comment (ymesika) [inline operator/cmd/mesh/install.go:186]: This error is going to be some kube client error.
Do we want to wrap/hide it with a user friendly error message?

Comment (hanxiaop) [reply operator/cmd/mesh/install.go:186]: Makes sense, the error could be more informative, thanks!

**Review by ymesika — COMMENTED** [review]
LG, thanks

File: releasenotes/notes/48982.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+issue:
+- https://github.com/istio/istio/issues/48643
+releaseNotes:
+- |
+  **Added** a message to indicate the default webhook shifting from a revisioned installation to a default installation.
```
Comment (linsun) [inline releasenotes/notes/48982.yaml:8]: specify which webhook? is it for mutating and validation or both?

Also, any impact for ambient?

**Comment by linsun** [comment]
> Sample out about a revisioned installation following a default installation:
> 
> ```
> ➜ istioctl install -f out/yamls/overlay.yaml --revision=test-rev
> This will install the Istio 1.21.0 "ambient" profile (with components: Istio core, Istiod, CNI, and Ztunnel) into the cluster. Proceed? (y/N) y
> ✔ Istio core installed                                                                                                                                                                
> ✔ Istiod installed                                                                                                                                                                    
> ✔ CNI installed                                                                                                                                                                       
> ✔ Ztunnel installed                                                                                                                                                                   
> ✔ Installation complete                                                                                                                                                               Made this installation the default for injection and validation.
> ➜ istioctl install -f out/yamls/overlay.yaml                    
> This installation will make default injection and validation pointing to the default revision, and originally it was pointing to the revisioned one.
> This will install the Istio 1.21.0 "ambient" profile (with components: Istio core, Istiod, CNI, and Ztunnel) into the cluster. Proceed? (y/N) 
> ```

Worth specify this is for sidecars?

Also, the output seems different from your steps - `revisioned installation following a default installation`?

**Review by linsun — COMMENTED** [review]
I don't fully understand the outputs, partly because I don't understand the steps you described clearly.

**Comment by hanxiaop** [comment]
> Worth specify this is for sidecars?

It's not related to sidecar or ambient, it's related to installation and revisions.

> Also, the output seems different from your steps - `revisioned installation following a default installation` - do you mean you have revisioned install first then default?

Correct.

File: releasenotes/notes/48982.yaml (hunk 2 above)
Comment (linsun) [reply releasenotes/notes/48982.yaml:8]: Ambient does not using mutating webhook, which is why I asked.

**Review by linsun — APPROVED** [review]
LGTM

**Pull request merged**